evm/v0.5/testv2/Owned.test.ts

```diff
@@ -0,0 +1,130 @@
+import * as h from '../src/helpers'
+import { OwnedTestHelperFactory } from '../src/generated'
+import { makeTestProvider } from '../src/provider'
+import { Instance } from '../src/contract'
+import { ethers } from 'ethers'
+import { assert } from 'chai'
+
+const ownedTestHelperFactory = new OwnedTestHelperFactory()
+const provider = makeTestProvider()
+
+let personas: h.Personas
+let owner: ethers.Wallet
+let nonOwner: ethers.Wallet
+let newOwner: ethers.Wallet
+
+beforeAll(async () => {
+  const rolesAndPersonas = await h.initializeRolesAndPersonas(provider)
+  personas = rolesAndPersonas.personas
+  owner = personas.Carol
+  nonOwner = personas.Neil
+  newOwner = personas.Ned
+})
+
+describe('Owned', () => {
+  let owned: Instance<OwnedTestHelperFactory>
+  const ownedEvents = ownedTestHelperFactory.interface.events
+
+  beforeEach(async () => {
+    owned = await ownedTestHelperFactory.connect(owner).deploy()
+  })
+
+  it('has a limited public interface', () => {
+    h.checkPublicABI(ownedTestHelperFactory, [
+      'acceptOwnership',
+      'owner',
+      'transferOwnership',
+      // test helper public methods
+      'modifierOnlyOwner',
+    ])
+  })
+
+  describe('#constructor', () => {
+    it('assigns ownership to the deployer', async () => {
+      const [actual, expected] = await Promise.all([
+        owner.getAddress(),
+        owned.owner(),
+      ])
+
+      assert.equal(actual, expected)
+    })
+  })
+
+  describe('#onlyOwner modifier', () => {
+    describe('when called by an owner', () => {
+      it('successfully calls the method', async () => {
+        const tx = await owned.connect(owner).modifierOnlyOwner()
+        const receipt = await tx.wait()
+
+        expect(h.findEventIn(receipt, ownedEvents.Here)).toBeDefined()
+      })
+    })
+
+    describe('when called by anyone but the owner', () => {
+      it('reverts', async () => {
+        // TODO: Revamp our revert checker to allow to check for revert messages
```

Comment:
I added a [chore for this](https://www.pivotaltracker.com/story/show/170583244). Mind cutting the TODOs? I think this can be better prioritized in Tracker.